src/microbit/__model/microbit_model.py

```diff
@@ -25,18 +28,64 @@ def running_time(self):
         print(f"time. time: {time.time()}")
         return time.time() - self.__start_time
 
+    def temperature(self):
+        return self.__temperature
+
+    def __set_temperature(self, temperature):
+        if (
+            temperature < CONSTANTS.MIN_TEMPERATURE
+            or temperature > CONSTANTS.MAX_TEMPERATURE
+        ):
+            raise ValueError(CONSTANTS.INVALID_TEMPERATURE_ERR)
+        else:
+            self.__temperature = temperature
+
     def update_state(self, new_state):
+        self.__update_buttons(new_state)
+        self.__update_motion(new_state)
+        self.__update_light(new_state)
+        self.__update_temp(new_state)
+
+    # helpers
+    def __update_buttons(self, new_state):
+        # get button pushes
         for button_name in CONSTANTS.EXPECTED_INPUT_BUTTONS:
-            button = self.microbit_button_dict[button_name]
+            button = self.__microbit_button_dict[button_name]
 
-            previous_pressed = button.is_pressed()
-            button_pressed = new_state.get(button_name, previous_pressed)
+            was_button_pressed = button.is_pressed()
+            is_button_pressed = new_state.get(button_name, was_button_pressed)
 
-            if button_pressed != previous_pressed:
-                if button_pressed:
+            if is_button_pressed != was_button_pressed:
+                if is_button_pressed:
                     button._Button__press_down()
                 else:
                     button._Button__release()

```

Comment:
Kind of following up from last PR, I would put this logic inside the components.
In this function you would only have:
```
for button_name in CONSTANTS.EXPECTED_INPUT_BUTTONS:
    if button_name in new_state:
        self.__microbit_button_dict[button_name].__update(new_state[buttom_name])
```
All the state changing logic would be inside `__update()`.
Same for the other components (accelerometer, temperature, etc.).
Up to you on how you want to proceed, just wanted to make sure I was clear on what I meant.

Reply:
I see! I incorporated this into my latest commits :). Only the temperature is not abstracted because the temperature value is stored in the microbit model itself.